### ISSUE TITLE
Generating Dynaconf object from a python dict

### ISSUE THREAD

**Issue by pvij** [dynaconf/dynaconf#712]
**Is your feature request related to a problem? Please describe.**
I usually use JSON configs in python projects and upload them to AWS parameter store. I really like using dynaconf as it helps me in validating the config and converting the config to an object. The problem I face is that when I get the parameter from the AWS parameter store, I get a stringified JSON (which has to be extracted from the response). Now, to use dynaconf, I have to write the stringified JSON to a file and then use dynaconf. It seems to me that writing to a file is an unnecessary additional step.

**Describe the solution you'd like**
I would like a Dynaconf object to be created from a python dict structure.

**Describe alternatives you've considered**
* I can simply stop using dynaconf and use the dict directly (and try validating the dict on my own).

**Additional context**
* I tried searching for any similar issue that might already exist but I didn't find any. Please let me know if there is a similar issue I might have missed.
* If there is any advice you have regarding the way I am getting the config, please feel free to share it. Also, if my request doesn't make sense, it would be great if you could point out why with reason.

Thank you for reading the above.

**Comment by rochacbruno** [comment]
Did you mean something like?

```py
from dynaconf import Dynaconf
settings = Dynaconf()

data = {"name": "Bruno", "info__age": 30, "info__id": 123}
settings.update(data)

In [6]: settings.name
Out[6]: 'Bruno'

In [7]: settings.info
Out[7]: <Box: {'id': 123, 'age': 30}>

In [8]: settings.info.id
Out[8]: 123

In [9]: settings.info.age
Out[9]: 30



```

**Comment by pvij** [comment]
Got it.
```
from dynaconf import Dynaconf, Validator
config = Dynaconf() 
config.update(data, validators=[Validator("name", must_exist=True)])
```
Thank you for getting back to me :-)

**Comment by rochacbruno** [comment]
maybe

```py
config.validators.register(Validator("aws", "elasticsearch", must_exist=True))
config.update(data)
config.validators.validate()
```

validators are not trigerred upon update, maybe they should... we can change it.

**Comment by pvij** [comment]
Will have to go deeper into the source code (and also see how other similar libraries work) to comment if it should be changed. But the above works for now. Thank you for your replies.

**Comment by rochacbruno** [comment]
We can add `validate=True` to methods `update`, `set`, `load_file`.

All the methods that can be called on-demand, so it is optional.

**Comment by pedro-psb** [comment]
> We can add validate=True to methods update, set, load_file.
> 
> All the methods that can be called on-demand, so it is optional.

It would be good to add a global option too (like `validate_on_update`) to set the default. This would allow the feature to be added without breaking behaviour (by setting it to False until 4.0.0)

**Comment by pedro-psb** [comment]
Closed by #900
